ballerina/http_connection.bal

```diff
@@ -266,8 +277,9 @@ isolated function setPayload(anydata payload, Response response, string? mediaTy
 isolated function processAnydata(Response response, anydata payload, string? mediaType = (), boolean setETag = false) {
     match mediaType {
         mime:APPLICATION_FORM_URLENCODED => {
-            if payload is map<string> {
-                string|error result = retrieveUrlEncodedData(payload);
+            map<string>|error pairs = checkpanic val:cloneWithType(payload);
```

Comment:
Here once you do the `checkpanic`, there wont be any error left as a return type right?